### PR TITLE
Backport apache/incubator-beam#1238

### PR DESCRIPTION
**Pull request by sammcveety**
R: @dhalperi

### CODE REVIEW

**Comment by sammcveety** [comment]
Rebased.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java [hunk 1]
```diff
@@ -710,14 +797,31 @@ public void populateDisplayData(DisplayData.Builder builder) {
...
        * Returns the query to be read, or {@code null} if reading from a table instead.
        */
       public String getQuery() {
+        return query == null ? null : query.get();
+      }
+
+      /**
+       * Returns the query to be read, or {@code null} if reading from a table instead.
+       */
+      public ValueProvider<String> getQueryProivder() {
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java]: [add `@Nullable` above, while you're at it]

**Comment by dhalperi** [comment]
LGTM with minor but important fixes

**Comment by dhalperi** [comment]
Thanks -- and please forward-port the fixups to Beam :)

**Pull request merged**